server/src/test/java/io/crate/planner/operators/EquiJoinDetectorTest.java

```diff
@@ -119,4 +119,25 @@ public void test_equality_expression_followed_by_case_expression() {
         Symbol joinCondition = sqlExpressions.asSymbol("t1.a = t1.a AND CASE 1 WHEN t1.a THEN false ELSE t2.b in (t2.b) END");
         assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition)).isFalse();
     }
+
+    @Test
+    public void equality_and_many_relations_in_boolean_join_condition_hash_join_not_possible() {
+        Symbol joinCondition = sqlExpressions.asSymbol("(t1.a >= 1) = ((t1.a = t1.a) AND (t2.b <= t2.b))");
```

Comment:
maybe for future proof add on more test with more levels of nested `=`, e.g.:
```
expr1 = expr2 AND (expr3 = (t1.a = t1.a + 10)  AND (expr 5 = t1.b = t1.b - 5)
```

Reply:
mismatched input '<' expecting {<EOF>, 'AT', 'OR', 'AND', '+', '-', '*', '^', '/', '%', '||', '::', '.', '[', '&', '|', '#'}

Tried different versions, having this exception. Will check, probably we cannot even do nested nested equal.

Reply:
Ok, had to wrap each side in extra parenthesis. Same in PG

Reply:
yep, sorry, I wrote this in a hurry, thx for figuring out.